mlir/test/mlir-miopen-lib/populate_bwd.mlir

```diff
@@ -1,7 +1,7 @@
-// RUN: mlir-miopen-lib-test --args " --operation conv2d_bwd_data --arch gfx906 --num_cu 64 --fil_layout NCHW --in_layout NCHW --out_layout NCHW --batchsize 64 --in_channels 1024 --out_channels 1024 --in_h 14 --in_w 14 --out_h 14 --out_w 14 --fil_h 1 --fil_w 1 --dilation_h 1 --dilation_w 1 --conv_stride_h 1 --conv_stride_w 1 --padding_h 0 --padding_w 0 --kernel_name conv2d_bwd" --option cflags | FileCheck %s --check-prefix=CFLAGS
-// RUN: mlir-miopen-lib-test --args " --operation conv2d_bwd_data --arch gfx906 --num_cu 64 --fil_layout NCHW --in_layout NCHW --out_layout NCHW --batchsize 64 --in_channels 1024 --out_channels 1024 --in_h 14 --in_w 14 --out_h 14 --out_w 14 --fil_h 1 --fil_w 1 --dilation_h 1 --dilation_w 1 --conv_stride_h 1 --conv_stride_w 1 --padding_h 0 --padding_w 0 --kernel_name foo" --option source | FileCheck %s --check-prefix=SOURCE
-// RUN: mlir-miopen-lib-test --args " --operation conv2d_bwd_data --arch gfx906 --num_cu 64 --fil_layout NCHW --in_layout NCHW --out_layout NCHW --batchsize 64 --in_channels 1024 --out_channels 1024 --in_h 14 --in_w 14 --out_h 14 --out_w 14 --fil_h 1 --fil_w 1 --dilation_h 1 --dilation_w 1 --conv_stride_h 1 --conv_stride_w 1 --padding_h 0 --padding_w 0 --kernel_name bar" --option header | FileCheck %s --check-prefix=HEADER
-// RUN: mlir-miopen-lib-test --args " --operation conv2d_bwd_data --arch gfx906 --num_cu 64 --fil_layout NCHW --in_layout NCHW --out_layout NCHW --batchsize 64 --in_channels 1024 --out_channels 1024 --in_h 14 --in_w 14 --out_h 14 --out_w 14 --fil_h 1 --fil_w 1 --dilation_h 1 --dilation_w 1 --conv_stride_h 1 --conv_stride_w 1 --padding_h 0 --padding_w 0 --kernel_name conv2d_nchw_nchw_nchw" --option bin | FileCheck %s --check-prefix=BIN
+// RUN: mlir-miopen-lib-test --args " --operation conv2d_bwd_data --arch gfx906 --num_cu 64 --in_type fp32 --fil_type fp32 --out_type fp32 --fil_layout NCHW --in_layout NCHW --out_layout NCHW --batchsize 64 --in_channels 1024 --out_channels 1024 --in_h 14 --in_w 14 --out_h 14 --out_w 14 --fil_h 1 --fil_w 1 --dilation_h 1 --dilation_w 1 --conv_stride_h 1 --conv_stride_w 1 --padding_h 0 --padding_w 0 --kernel_name conv2d_bwd" --option cflags | FileCheck %s --check-prefix=CFLAGS
```

Comment:
Thanks for filing this hole!

Should we allow mixed type between input/output/filter? I think one type is enough.

Reply:
It only adds a conversion, may be useful someday, maybe for outputs of a lower precision.

Reply:
Sounds good!